### PR TITLE
Common definitions

### PR DESCRIPTION
**Pull request by thibault-martinez**
This PR 
- removes duplicate definitions of the same values
- renames values for clarity
- relocates all common definitions under `/common/defs.h`

# Test Plan:
CI

### CODE REVIEW

**Comment by th0br0** [comment]
Let's hold back on merging this until #297 is done as that also takes care of some #defines

File: common/defs.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#define NUMBER_OF_TRITS_IN_A_BYTE 5
+
+#define TRYTE_VALUE_MAX 13
+#define TRYTE_VALUE_MIN -13
+#define TRYTE_WIDTH 3
+
+#define HASH_LENGTH_BYTE 48
+#define HASH_LENGTH_TRYTE 81
+#define HASH_LENGTH_TRIT 243
+
```
Comment (tsvisabo) [inline common/defs.h:30]: can you leave comments to distinguish between sections?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: common/defs.h (hunk 1 above)
Comment (thibault-martinez) [reply common/defs.h:30]: Sure

File: common/curl-p/hamming.c [hunk 2]
```diff
@@ -18,7 +19,8 @@ short test(PCurl *const curl, unsigned short const security) {
     sum = 0;
 
     for (j = 0; j < security; j++) {
-      for (k = j * HASH_LENGTH / 3; k < (j + 1) * HASH_LENGTH / 3; k++) {
+      for (k = j * HASH_LENGTH_TRIT / 3; k < (j + 1) * HASH_LENGTH_TRIT / 3;
```
Comment (th0br0) [inline common/curl-p/hamming.c]: While we're at it, we should also replace the various `HASH_LENGTH_TRIT / 3` with `HASH_LENGTH_TRYTE`

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/curl-p/hamming.c (hunk 2 above)
Comment (thibault-martinez) [reply common/curl-p/hamming.c]: Will do

**Review by th0br0 — APPROVED** [review]
Let's get this merged so it doesn't lag behind too much & we can reduce our backlog :)

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**